library/printoptions.mli

```diff
@@ -0,0 +1,78 @@
+(** Printing options record *)
+
+(* storage of the options is in the current instance of the record *)
+
+type t = {
+  printing_allow_match_default_clause : bool;
+  printing_coercions : bool;
+  printing_compact_contexts : bool;
+  printing_existential_instances : bool;
+  printing_factorizable_match_patterns : bool;
+  printing_implicit : bool;
+  printing_implicit_defensive : bool;
+  printing_matching : bool;
+  printing_notations : bool;
+  printing_primitive_projection_compatibility : bool;
+  printing_primitive_projection_parameters : bool;
+  printing_projections : bool;
+  printing_records : bool;
+  printing_synth : bool;
+  printing_universes : bool;
+  printing_wildcard : bool;
+}
+
+(** getter/setter for Printing options *)
+val get_current_options : unit -> t
+val set_current_options : t -> unit
+
+(** getter/setters for Printing options *)
+val set_printing_all : unit -> unit
+val set_printing_sugared : unit -> unit
+val set_printing_defaults : unit -> unit
+
+(** for any set of options, return all-enabled options, useful for
+    `with_printing_option` below
+ *)
+val get_printing_all : t -> t
```

Comment:
This type is kinda strange/non uniform, I would prefer having `val printing_all : t` and let people write `fun _ => printing_all`.

Reply:
In fact, I did it that way first. I'll go back to that.